paramiko/client.py

```diff
@@ -608,17 +608,27 @@ def _auth(self, username, password, pkey, key_filenames, allow_agent,
                     )
                     if os.path.isfile(full_path):
                         keyfiles.append((keytype, full_path))
+                        if os.path.isfile(full_path + '-cert.pub'):
+                            keyfiles.append((keytype, full_path + '-cert.pub'))
 
             if not look_for_keys:
                 keyfiles = []
 
             for pkey_class, filename in keyfiles:
                 try:
-                    key = pkey_class.from_private_key_file(filename, password)
-                    self._log(
-                        DEBUG,
-                        'Trying discovered key %s in %s' % (
-                            hexlify(key.get_fingerprint()), filename))
+                    if filename.endswith('-cert.pub'):
+                        key = pkey_class.from_private_key_file(filename.rstrip('-cert.pub'), password)
```

Comment:
`rstrip()` is not what you want to use here - the argument is not a suffix, it is a set of characters.

```
>>> "abc-cert.pub".rstrip('-cert.pub')
'a'
```

instead you could do something like `filename[:-len("-cert.pub")]`